archive/100-theorems-list/37_solution_of_cubic.lean

```diff
@@ -0,0 +1,199 @@
+/-
+Copyright (c) 2021 Jeoff Lee. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Jeoff Lee
+-/
+import tactic.basic
+import data.complex.basic
+import algebra.group.defs
+
+/-!
+# The Solution of Cubic
+
+This file proves Theorem 37 from the [100 Theorems List](https://www.cs.ru.nl/~freek/100/).
+
+The theorem `cubic_eq_zero_iff` gives solution to the cubic equation over ℂ,
+`cubic_eq_zero_iff_of_discrim_eq_zero`
+based on the [Cardano's Formula](https://en.wikipedia.org/wiki/Cubic_equation#Cardano's_formula).
+
+## References
+
+Originally ported from Isabelle/HOL. The
+[original file](https://isabelle.in.tum.de/dist/library/HOL/HOL-ex/Cubic_Quartic.html) was written by Amine Chaieb.
+
+## Tags
+polynomial, cubic, root
+-/
+
+section
+
+open complex
+
+variables (a b c d : ℂ)
+variables {p q r s t : ℂ}
+
+noncomputable def ω : ℂ := ⟨-1/2, (real.sqrt 3)/2⟩
+
+lemma omega_cube_eq_one : ω ^ 3 = 1 :=
```

Comment:
You won't need this in the approach I'm suggesting, because you will just `is_primitive_root.pow_eq_one`.